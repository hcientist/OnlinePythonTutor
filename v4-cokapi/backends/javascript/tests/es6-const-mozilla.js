// From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const

// define MY_FAV as a constant and give it the value 7
const MY_FAV = 7;

// will print 7
console.log("my favorite number is: " + MY_FAV);

// MY_FAV is still 7
console.log("my favorite number is " + MY_FAV);

// Assigning to A const variable is a syntax error
const A = 1;

// const also works on objects
const MY_OBJECT = {"key": "value"};

// Overwriting the object fails as above (in Firefox and Chrome but not in Safari)
MY_OBJECT = {"OTHER_KEY": "value"};
