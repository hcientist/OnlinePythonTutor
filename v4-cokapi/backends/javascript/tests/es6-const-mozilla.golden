{ code: '// From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const\n\n// define MY_FAV as a constant and give it the value 7\nconst MY_FAV = 7;\n\n// will print 7\nconsole.log("my favorite number is: " + MY_FAV);\n\n// MY_FAV is still 7\nconsole.log("my favorite number is " + MY_FAV);\n\n// Assigning to A const variable is a syntax error\nconst A = 1;\n\n// const also works on objects\nconst MY_OBJECT = {"key": "value"};\n\n// Overwriting the object fails as above (in Firefox and Chrome but not in Safari)\nMY_OBJECT = {"OTHER_KEY": "value"};',
  trace: 
   [ { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: {},
       ordered_globals: [],
       line: 4,
       col: 0,
       event: 'step_line',
       heap: {} },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { MY_FAV: 7 },
       ordered_globals: [ 'MY_FAV' ],
       line: 7,
       col: 0,
       event: 'step_line',
       heap: {} },
     { stdout: 'my favorite number is: 7\n',
       func_name: '',
       stack_to_render: [],
       globals: { MY_FAV: 7 },
       ordered_globals: [ 'MY_FAV' ],
       line: 10,
       col: 0,
       event: 'step_line',
       heap: {} },
     { stdout: 'my favorite number is: 7\nmy favorite number is 7\n',
       func_name: '',
       stack_to_render: [],
       globals: { MY_FAV: 7 },
       ordered_globals: [ 'MY_FAV' ],
       line: 13,
       col: 0,
       event: 'step_line',
       heap: {} },
     { stdout: 'my favorite number is: 7\nmy favorite number is 7\n',
       func_name: '',
       stack_to_render: [],
       globals: { MY_FAV: 7, A: 1 },
       ordered_globals: [ 'MY_FAV', 'A' ],
       line: 16,
       col: 0,
       event: 'step_line',
       heap: {} },
     { stdout: 'my favorite number is: 7\nmy favorite number is 7\n',
       func_name: '',
       stack_to_render: [],
       globals: { MY_FAV: 7, A: 1, MY_OBJECT: [ 'REF', 1 ] },
       ordered_globals: [ 'MY_FAV', 'A', 'MY_OBJECT' ],
       line: 19,
       col: 0,
       event: 'step_line',
       heap: { '1': [ 'INSTANCE', '', [ 'key', 'value' ] ] } },
     { exception_msg: 'TypeError: Assignment to constant variable.',
       stdout: 'my favorite number is: 7\nmy favorite number is 7\n',
       func_name: '',
       stack_to_render: [],
       globals: { MY_FAV: 7, A: 1, MY_OBJECT: [ 'REF', 1 ] },
       ordered_globals: [ 'MY_FAV', 'A', 'MY_OBJECT' ],
       line: 19,
       col: 10,
       event: 'exception',
       heap: { '1': [ 'INSTANCE', '', [ 'key', 'value' ] ] } } ] }
